lib/rubocop/cop/style/redundant_regexp_character_class.rb

```diff
@@ -104,11 +104,10 @@ def backslash_b?(elem)
           elem == '\b'
         end
 
-        def backslash_zero?(elem)
-          # See https://github.com/rubocop/rubocop/issues/11067 for details - in short "\0" != "0" -
-          # the former means an Unicode code point `"\u0000"`, the latter a number character `"0"`.
-          # Similarly "\032" means "\u001A". Other numbers starting with "\0" can also be mentioned.
-          elem == '\0'
+        def octal_requiring_char_class?(elem)
+          # The octal escapes \1 to \7 only work inside a character class
+          # because they would be a backreference outside it.
+          elem.match?(/\A\\[1-7]\z/)
         end
```

Comment:
As far as I remember `add_runtime_dependency` cannot be updated to `>= 2.7`: https://github.com/rubocop/rubocop/pull/9154 
Isn't backwards compatibility like below needed?

```suggestion
        if Gem::Version.new(Regexp::Parser::VERSION) >= Gem::Version.new('2.7')
          def octal_requiring_char_class?(elem)
            # The octal escapes \1 to \7 only work inside a character class
            # because they would be a backreference outside it.
            elem.match?(/\A\\[1-7]\z/)
          end
        else
          def octal_requiring_char_class?(elem)
           # See https://github.com/rubocop/rubocop/issues/11067 for details - in short "\0" != "0" -
           # the former means an Unicode code point `"\u0000"`, the latter a number character `"0"`.
           # Similarly "\032" means "\u001A". Other numbers starting with "\0" can also be mentioned.
           elem == '\0'
          end
        end
```

Reply:
no, because `\0` (or anything starting with `\0`) acts the same way inside and outside a char class.

basically, this commit fixes both the spec (which was only green due to a bug in regexp_parser), as well as the behavior.

before this commit:

```ruby
/[\0]/   # no offense (wrong - this char class *is* redundant)
/[\032]/ # no offense (wrong - this char class *is* redundant)
/[\1]/   # offense (wrong - this char class *is* needed)
/[\132]/ # no offense (wrong - this char class *is* redundant)
/[\9]/   # offense (correct - no octal, so both char class and escaping are redundant)
```

after:

```ruby
/[\0]/   # offense (correct)
/[\032]/ # no offense on regexp_parser 2.6 (still wrong), offense on 2.7 (correct)
/[\1]/   # no offense (correct)
/[\132]/ # no offense (correct)
/[\9]/   # offense (correct)
```

Reply:
I get it. Thank you for the explanation!